### ISSUE TITLE
Support for Wildfire (visual selection  expansion using text objects)

### ISSUE THREAD

**Issue by expelledboy** [VSCodeVim/Vim#2834]
Hey guys, love the idea of using vim in vscode!

Would it be possible atm to implement something similar to [vim's wildfire](https://github.com/gcmt/wildfire.vim) plugin? It expands the visual selection according to next recognized text objects, searching in order of a configurable list.

![Wildfire Preview](https://github.com/gcmt/wildfire.vim/blob/master/_assets/preview.gif?raw=true)

I use it so much I actually cant use vim anywhere that doesn't support it, just breaks my concentration.

**Comment by xconverge** [comment]
did you try our keybinding of ```af``` https://github.com/VSCodeVim/Vim#-vscodevim-tricks

You can rebind something else to af if that makes you more comfortable as well

**Comment by expelledboy** [comment]
@xconverge Wow! That was easy, you should add it to the readme!

```json
    "vim.normalModeKeyBindingsNonRecursive": [
        {
            "before": ["enter"],
            "after": ["v", "a", "f"],
        }
    ],
    "vim.visualModeKeyBindingsNonRecursive": [
        {
            "before": ["enter"],
            "after": ["a", "f"],
        }
    ]
```

The only other powerful feature lacking here is then, having custom rules for text object expansions per language. For example in Erlang I expand inside `{` character according to what its encapulating, but these do no apply to any other lanugage. Even the order matters, if we cant define our own custom text objects.

But now I am invested, I think I am going to have a read through how this all works. This project has come along way!

**Comment by Zyst** [comment]
@expelledboy I think it is, what do you think is missing there?

https://github.com/VSCodeVim/Vim#-vscodevim-tricks

If necessary it's possible to write more detailed docs for it.

**Comment by expelledboy** [comment]
@Zyst My bindings with `af` posted above is what I was looking for, but the final step to get the full wildfire goodness is to have custom rules for text expansion per filetype.

If anyone could point me to where its implemented I would appreciate it!

**Comment by cursorzz** [comment]
Is there any command that support select **Inside** block? for the example below

> af - visual mode command which selects increasingly large blocks of text. For example, if you had "blah (foo [bar 'ba|z'])" then it would select 'baz' first. If you pressed af again, it'd then select [bar 'baz'], and if you did it a third time it would select "(foo [bar 'baz'])".

I would like to select baz instead of 'baz'

**Comment by Glench** [comment]
Yes, is it possible to get `af` to work just like VSCode's `Expand Selection` command? It would be awesome if they were consistent.

**Comment by d1y** [comment]
I checked and there is no such plugin in the whole community
But. maybe you can try this plugin 🤣
https://github.com/antfu/vscode-smart-clicks
https://twitter.com/antfu7/status/1506508822030200833

FYI. I have a mapped key that mimics this behavior as closely as possible

```json
    {
      "before": [
        "enter"
      ],
      "after": [
        "a",
        "f"
      ]
    }
```

**Comment by d1y** [comment]
FYI, I use https://github.com/jhasse/vscode-bracket-select the plugin

```jsonc
    {
      "before": [
        "enter"
      ],
      "commands": [ "bracket-select.select" ],
      // "after": [
      //   "v",
      //   "a",
      //   "f"
      // ]
    },
```

![demo.gif](https://s2.loli.net/2023/09/05/4uaSfvx9JLFdeA5.gif)

**Comment by expelledboy** [comment]
Amazing! Thanks @d1y you have made my day. I really missed the wildfire functionality
